new/analyse/spec/create-rlsr-json-content.spec.ts

```diff
@@ -0,0 +1,107 @@
+/* eslint-env node, jest */
+import type { Env, Module } from '../../types';
+
+import { envWithConfig } from '../../fixtures/env';
+
+import { createMockPackage } from '../fixtures/createMockPackage';
+
+// mock Logger
+const mockError = jest.fn();
+const mockLog = jest.fn();
+const mockLogger = jest.fn(() => ({
+  error: mockError,
+  log: mockLog,
+}));
+jest.doMock('../../helpers/logger', () => ({
+  logger: mockLogger,
+}));
+
+describe('[analyse] createRlsrJsonContent module', () => {
+  let createRlsrJsonContent: Module;
+  beforeAll(() => {
+    createRlsrJsonContent =
+      require('../create-rlsr-json-content').createRlsrJsonContent;
+  });
+
+  it('throws an exception when "env.packages" is not defined', () => {
+    const expectedError = 'missing "packages" on env object';
+    expect(() => createRlsrJsonContent(envWithConfig)).toThrow(expectedError);
+    expect(mockError).toHaveBeenCalledTimes(1);
+    expect(mockError).toHaveBeenCalledWith(expectedError);
+  });
+
+  describe('on Run', () => {
+    let result: Env;
+
+    beforeAll(async () => {
+      const mockedEnv: Env = {
+        ...envWithConfig,
+        packages: {
+          package1: createMockPackage(
+            'package1',
+            {},
+            {},
+            {
+              currentVersion: '1.0.0',
+              incrementedVersion: '1.1.0',
+              dependsOn: [
+                {
+                  name: 'package2',
+                  range: '1-3',
+                  type: 'default',
+                },
+              ],
+            }
+          ),
+          package2: createMockPackage(
+            'package2',
+            {},
+            {},
+            {
+              currentVersion: '2.0.0',
+              dependsOn: [],
+            }
+          ),
+        },
+      };
+      result = await createRlsrJsonContent(mockedEnv);
```

Comment:
`createRlsrJsonContent` is not an async function, why is this needed?

Reply:
it returns a promise. I thought, it was the reason why a test failed (but probably wasn't). I change it (forgot about it).

Reply:
It's because of the Module type, I guess.
It's typed to return either an Env or a Promise<Env> object.

I think I've always forced it as `result = createRlsrJsonContent(mockedEnv) as Env;`